views/permission.go

```diff
@@ -121,5 +137,50 @@ func (v *Views) PermissionEditFunc(c echo.Context) error {
 
 // PermissionDeleteFunc handles a delete permission request
 func (v *Views) PermissionDeleteFunc(c echo.Context) error {
-	return nil
+	if c.Request().Method == http.MethodPost {
+		permissionID, err := strconv.Atoi(c.Param("permissionid"))
+		if err != nil {
+			log.Printf("failed to get permissionid for permission: %+v", err)
+			if !v.conf.Debug {
+				return echo.NewHTTPError(http.StatusInternalServerError, fmt.Errorf("failed to get permissionid for permission: %+v", err))
+			}
```

Comment:
It is if not debug, `!v.conf.Debug` and is there for when debugging happens and there is no db connected, this is so dev can keep going even if you are not on campus or even off the grid

Reply:
You can spin up a local database on your laptop, though - and even then, developing without a DB is a risk because you don't know if your queries are correct, so you'd need to test with a DB anyway, at which point you may as well use one at the start. Plus if the DB somehow drops out in production, you risk telling people that their changes have been applied when they really haven't.

What I'm trying to say is all this debug-only code provides minor short-term developer experience benefits, but at the cost of long-term pain (when your code breaks as soon as run against a database), added complexity (all those `if !v.conf.Debug`s everywhere), and risk (if an important condition is accidentally inside a `Debug` that's a production bug), so I'm really not sure it's worth it.

Reply:
Would you say to just remove these?

Reply:
In general yes, especially where the behaviour would be _functionally_ different between development and production.

Reply:
Fair enough, I'll remove it

Reply:
Fixed